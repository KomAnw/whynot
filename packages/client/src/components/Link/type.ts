import type { ReactNode } from 'react';

export type sizeType = 'size20' | 'size24' | 'size30' | 'size40';

export type propsLink = {
  to: string;
  children: ReactNode;
  variant?: sizeType;
  className?: string;
};

export type propsLinkStyle = Pick<propsLink, 'variant'>;
