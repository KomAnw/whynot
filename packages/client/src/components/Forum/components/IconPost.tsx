export const IconPost = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M10.67 19.9701C10.43 19.9901 10.19 20.0001 9.95 20.0001C8.78 20.0001 7.66999 19.7201 6.67999
      19.2301L2.78999 19.7901L3.38 15.8801H3.39C2.99 15.0001 2.75 14.0301 2.75 13.0001C2.75 11.1101 3.51999
      9.40014 4.75999 8.14014"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M10.8601 11.7802C11.3019 11.7802 11.6601 11.431 11.6601 11.0002C11.6601 10.5694 11.3019 10.2202
      10.8601 10.2202C10.4182 10.2202 10.0601 10.5694 10.0601 11.0002C10.0601 11.431 10.4182 11.7802 10.8601 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M14.0603 11.7802C14.5021 11.7802 14.8603 11.431 14.8603 11.0002C14.8603 10.5694 14.5021 10.2202
      14.0603 10.2202C13.6184 10.2202 13.2603 10.5694 13.2603 11.0002C13.2603 11.431 13.6184 11.7802 14.0603
      11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M17.2502 11.7802C17.692 11.7802 18.0502 11.431 18.0502 11.0002C18.0502 10.5694 17.692 10.2202
      17.2502 10.2202C16.8084 10.2202 16.4502 10.5694 16.4502 11.0002C16.4502 11.431 16.8084 11.7802 17.2502 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M20.6199 13.88L21.2099 17.79L17.3199 17.23C16.6199 17.58 15.8599 17.82 15.0499 17.93C14.7299
      17.98 14.3899 18 14.0499 18C10.0699 18 6.85986 14.87 6.85986 11C6.85986 9.07 7.64986 7.35 8.94986 6.08C10.2399
      4.79 12.0499 4 14.0599 4C18.0399 4 21.2499 7.13 21.2499 11C21.2499 12.03 21.0099 13 20.6099 13.88H20.6199V13.88Z"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
