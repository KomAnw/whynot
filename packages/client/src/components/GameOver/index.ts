export { GameOver as default } from './GameOver';
