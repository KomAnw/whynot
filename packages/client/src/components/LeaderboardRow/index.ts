export { LeaderboardRow as default } from './LeaderboardRow';
