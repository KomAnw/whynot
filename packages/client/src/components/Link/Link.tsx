import styled from 'styled-components';
import { Link as ReactLink } from 'react-router-dom';
import { sizeType, propsLink, propsLinkStyle } from 'components/Link/type';

function creatStyleProps(variant: sizeType): propsLinkStyle {
  switch (variant) {
    case 'small':
      return { fontSize: '20px', lineHeight: '27px' };
    case 'regular':
      return { fontSize: '40px', lineHeight: '47px' };
    case 'large':
      return { fontSize: '50px', lineHeight: '57px' };
    default:
      return { fontSize: '20px', lineHeight: '27px' };
  }
}

function Link({ children, url, variant }: propsLink) {
  const styleProps = creatStyleProps(variant);

  return (
    <StyledLink {...styleProps} to={url}>
      {children}
    </StyledLink>
  );
}

export default Link;

const StyledLink = styled(ReactLink)<propsLinkStyle>`
  cursor: pointer;
  text-decoration: none;
  &:hover {
    opacity: 0.8;
  }
  font-size: ${props => props.fontSize};
  line-height: ${props => props.lineHeight};
  font-family: ${({ theme }) => theme.fonts.main};
  color: ${({ theme }) => theme.colors.core.text.primary};
`;
