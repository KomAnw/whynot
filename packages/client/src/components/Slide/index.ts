export { SlideHistory, SlideTutorial } from './Slide';
