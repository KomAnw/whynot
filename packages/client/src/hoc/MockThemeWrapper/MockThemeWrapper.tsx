import { ThemeProvider } from 'styled-components';
import { useAppSelector } from 'src/hooks/redux';
import { ThemeWrapperProps } from './types';

const MockThemeWrapper = ({ children }: ThemeWrapperProps) => {
  const mockTheme = useAppSelector(({ mockTheme }) => mockTheme);

  return <ThemeProvider theme={mockTheme}>{children}</ThemeProvider>;
};

export default MockThemeWrapper;
