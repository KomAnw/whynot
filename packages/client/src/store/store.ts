import { configureStore } from '@reduxjs/toolkit';
import { authApi } from 'src/api/auth/auth';
import themeReducer from 'src/hoc/ThemeWrapper/themeSlice';
import mockThemeReducer from 'src/hoc/MockThemeWrapper/mockThemeSlice';

const store = configureStore({
  reducer: {
    theme: themeReducer,
    mockTheme: mockThemeReducer,
    [authApi.reducerPath]: authApi.reducer,
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(authApi.middleware),
});

export default store;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
