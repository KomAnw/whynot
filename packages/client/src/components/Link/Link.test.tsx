import { create } from 'react-test-renderer';
import { Link } from 'components/Link';
import { BrowserRouter } from 'react-router-dom';
import MockThemeWrapper from 'src/hoc/MockThemeWrapper';
import { Provider } from 'react-redux';
import store from 'src/store';
import 'jest-styled-components';

describe('Link', () => {
  it('renders <Link /> correctly', () => {
    const tree = create(
      <Provider store={store}>
        <BrowserRouter>
          <MockThemeWrapper>
            <Link to="/" variant="size24">
              LOGIN
            </Link>
          </MockThemeWrapper>
        </BrowserRouter>
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
