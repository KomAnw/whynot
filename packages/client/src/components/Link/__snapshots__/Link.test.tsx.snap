// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders <Link /> correctly 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 24px;
  line-height: 27px;
  font-family: Handjet;
  color: #6C5BC3;
}

.c0:hover {
  color: #6457b8b3;
}

.c0:hover {
  color: #6C5BC3;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
  variant="size24"
>
  LOGIN
</a>
`;
